### ISSUE TITLE
[1.9.1] NoSuchMethodException when trying to find the constructor for a simple data class.

### ISSUE THREAD

**Issue by andrewmunn** [square/moshi#1013]
Just upgraded to 1.9.1 and get this stacktrace when I try to run our app and parse the response:
```
java.lang.NoSuchMethodException: <init> [boolean, boolean, int, class kotlin.jvm.internal.DefaultConstructorMarker]
2019-11-11 09:07:51.024 25993-25993/? E/ContentStore$Companion: Failed to parse CurrentUserSession string
    java.lang.NoSuchMethodException: <init> [boolean, boolean, int, class kotlin.jvm.internal.DefaultConstructorMarker]
        at java.lang.Class.getConstructor0(Class.java:2327)
        at java.lang.Class.getDeclaredConstructor(Class.java:2166)
        at com.nextdoor.model.user.UserSettingsJsonAdapter.fromJson(UserSettingsJsonAdapter.kt:61)
        at com.nextdoor.model.user.UserSettingsJsonAdapter.fromJson(UserSettingsJsonAdapter.kt:19)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:40)
```

Here is the class in question:
```
@JsonClass(generateAdapter = true)
data class UserSettings(
    @Json(name = "nearby_neighborhoods_state") val nearbyNeighborhoodsState: Boolean? = null,
    @Json(name = "share_with_owner") val shareWithOwner: Boolean? = null
)
```

**Comment by ZacSweers** [comment]
Fixed on master, should be in the 1.9.2 release. Can you confirm by trying a master snapshot build?

**Comment by andrewmunn** [comment]
Confirmed fixed on 1.9.2
